idmanager/idmanager.go

```diff
@@ -3,51 +3,71 @@ package idmanager
 import (
 	"context"
 	"fmt"
+	"io"
 
-	"github.com/Scalingo/go-etcd-lock/lock"
+	etcdlock "github.com/Scalingo/go-etcd-lock/lock"
 	"github.com/Scalingo/sand/config"
 	"github.com/Scalingo/sand/etcd"
 	"github.com/Scalingo/sand/store"
 	"github.com/pkg/errors"
 )
 
 type Manager interface {
-	Lock(context.Context) error
+	Lock(context.Context) (Lock, error)
 	Generate(context.Context) (int, error)
-	Unlock(context.Context) error
+}
+
+type Lock interface {
+	Unlock(ctx context.Context) error
 }
 
 type manager struct {
 	store  store.Store
 	config *config.Config
-	lock   lock.Lock
 	field  string
 	name   string
 	prefix string
 }
 
+type lock struct {
+	resourceLock   etcdlock.Lock
+	lockingBackend io.Closer
+}
+
 func New(c *config.Config, s store.Store, name string, field string, prefix string) Manager {
 	return &manager{config: c, store: s, field: field, name: name, prefix: prefix}
 }
 
-func (m *manager) Lock(ctx context.Context) error {
+func (m *manager) Lock(ctx context.Context) (Lock, error) {
+	var l Lock
+
 	client, err := etcd.NewClient()
```

Comment:
Here we were creating a new client, which was never Close(), so we were leaking connections to etcd each time a new ID was generated

Reply:
The goal of this PR is to close correctly the client after usage

Reply:
Thanks !